### PR TITLE
[WIP] Model bindings: Abstracted out findFirst method. Set params from cache to internalCache class property.

### PR DESCRIPTION
**Pull request by linxlad**
Hello!

- [x] Type: bug fix | new feature

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.

Small description of change:

Added a new `findBoundModel` method for extendability. Params fetched from cache are being added to internalCache class property in Phalcon\Mvc\Model\Binder:getParamsFromCache.

Thanks
Nathan Daly <justlikephp@gmail.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # [3.2.0](https://github.com/phalcon/cphalcon/releases/tag/v3.2.0) (2017-XX-XX)
+- Added a new `Phalcon\Mvc\Model\Binder::findBoundModel` method. Params fetched from cache are being added to `internalCache`  class property in `Phalcon\Mvc\Model\Binder::getParamsFromCache`.
```
Comment (Jurigag) [inline CHANGELOG.md]: Ad it add the end of this tag.

**Comment by Jurigag** [comment]
Also squash commits.

**Comment by linxlad** [comment]
Commits squashed and CHANGELOG.md adjusted accordingly.

**Comment by sergeyklay** [comment]
@Jurigag Could you please take a look

File: phalcon/mvc/model/binder.zep [hunk 2]
```diff
@@ -172,18 +184,18 @@ class Binder implements BinderInterface
...
 				if typeof realClasses == "array" {
 					if fetch className, realClasses[paramKey] {
-						let boundModel = {className}::findFirst(paramValue);
+						let boundModel = this->findBoundModel(paramValue, className);
 					} else {
 						throw new Exception("You should provide model class name for ".paramKey." parameter");
 					}
 				} elseif typeof realClasses == "string" {
-					let boundModel = {realClasses}::findFirst(paramValue);
+					let boundModel = this->findBoundModel(paramValue, className);
```
Comment (Jurigag) [inline phalcon/mvc/model/binder.zep]: It should be realClasses here.

**Review by sergeyklay — APPROVED** [review]
Looks good to me

**Comment by sergeyklay** [comment]
Cc: @phalcon/framework-team

**Review by Jurigag — APPROVED** [review]

**Comment by virgofx** [comment]
Looks good.

**Comment by sergeyklay** [comment]
Thank you

**Comment by Jurigag** [comment]
Well, it was WIP because he was planning to add more here, but whatever

**Comment by linxlad** [comment]
I will raise the next part as a new feature, with model bindings

**Comment by sergeyklay** [comment]
@lantian Could you please reflect these changes in docs with small note/tip?

**Pull request merged**